Functions/functions_collections_home_task_edited.py

```diff
@@ -0,0 +1,81 @@
+"""
+Home task for Module 4 - Functions (Based on Module 2 - Collections home task)
+
+Write a code, which will:
+
+1. create a list of random number of dicts (from 2 to 10)
+
+dict's random numbers of keys should be letter,
+dict's values should be a number (0-100),
+example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+2. get previously generated list of dicts and create one common dict:
+
+if dicts have same key, we will take max value, and rename key with dict number with max value
+if key is only in one dict - take it as is,
+example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+Each line of code should be commented with description.
+
+Commit script to git repository and provide link as home task result.
+"""
+import random
+import string
+
+
+# Function for generation of single dictionary with defined ranges of random keys and values
+def generate_dictionary_with_random_keys_and_values(
+        set_of_keys,
+        start_of_value, end_of_value,
+        start_of_total_count, end_of_total_count,
+):
+    final_dictionary = {random.choice(set_of_keys): random.randint(start_of_value, end_of_value)
+                        for _ in range(random.randint(start_of_total_count, end_of_total_count))}
+    return final_dictionary
+
+
+# Function for generation of list with defined count of random generated dictionaries
+def create_list_with_random_dictionaries(start_of_count_of_dicts, end_of_count_of_dicts):
```

Comment:
```suggestion
def create_random_dictionaries_list(min_dicts_count, max_dicts_count, keys, min_value, max_value, min_values_count, max_values_count):
"""
some docstring
"""
```
You pass (string.ascii_lowercase, 0, 100, 1, 100) but it's not mentioned in name of function, may be you should also pass these values as arguments? In this case if you want to use this function for generatin random dictionaries with another keys and values you can do so.

Reply:
Fixed.